src/TempleLineOfCredit.sol

```diff
@@ -0,0 +1,318 @@
+        return positions[account].debtAmount;
+    }
+
+    /**
+     * @dev Get user total debt incurred (principal + interest)
+     * @return total Debt
+     */
+    function getTotalDebtAmount(address account) public view returns (uint256) {
+        uint256 totalDebt = positions[account].debtAmount;
+        uint256 periodsPerYear = 365 days / interestRatePeriod;
+        uint256 periodsElapsed = block.timestamp - positions[account].createdAt; // divided by interestRatePeriod
+        totalDebt += (((totalDebt * interestRateBps) / 10000 / periodsPerYear) * periodsElapsed) / interestRatePeriod;
+        return totalDebt;
+    }
+
+    /**
+     * @dev Allows operator to depoist debt tokens
+     * @param amount is the amount to deposit
+     */
+    function depositDebt(address account, uint256 amount) external onlyOperators{
+        if (amount == 0) {
+            revert InvalidAmount(amount);
+        }
+        debtBalance += amount;
+        debtToken.safeTransferFrom(
+            account,
+            address(this),
+            amount
+        );
+        emit DepositDebt(amount);
+    }
+
+        /**
+     * @dev Allows operator to remove debt token
+     * @param amount is the amount to remove
+     */
+    function removeDebt(uint256 amount) external onlyOperators{
+        if (amount == 0) {
+            revert InvalidAmount(amount);
+        }
+        debtBalance -= amount;
+        debtToken.safeTransfer(
+            msg.sender,
+            amount
+        );
+        emit RemoveDebt(amount);
+    }
+
+    /**
+     * @dev Allows borrower to deposit collateral
+     * @param amount is the amount to deposit
+     */
+    function postCollateral(uint256 amount) external {
+        if (amount == 0) revert InvalidAmount(amount);
+        positions[msg.sender].collateralAmount += amount;
+        collateralToken.safeTransferFrom(
+            msg.sender,
+            address(this),
+            amount
+        );
+        emit PostCollateral(msg.sender, amount);
+    }
+
+    function borrow(uint256 amount) external {
+        if (amount == 0) revert InvalidAmount(amount);
+
+        uint256 debtAmount =  positions[msg.sender].debtAmount;
+        if (debtAmount != 0) {
+            debtAmount = getTotalDebtAmount(msg.sender);
+        }
+
+        uint256 borrowCapacity = _maxBorrowCapacity(positions[msg.sender].collateralAmount) - debtAmount;
+        debtAmount += amount;
+
+        if (debtAmount > borrowCapacity) {
+            revert InsufficentCollateral(borrowCapacity, debtAmount);
+        }
+
+        positions[msg.sender].debtAmount = debtAmount;
+        
+        // If more than 1 interest rate period has passed update the start-time
+        if (block.timestamp - positions[msg.sender].createdAt >= interestRatePeriod || positions[msg.sender].createdAt == 0 ) {
+            positions[msg.sender].createdAt = block.timestamp;
+        }
+         
+        debtBalance -= amount;
```

Comment:
This will revert if the amount to borrow is more than the DAI tokens in `debtBalance` (ie the operator needs to depositDebt() to add more DAI). I think it would be nicer UX (for the user) to revert with a custom error at the top of this function, ie fail early

Reply:
good point